packages/flutter_tools/lib/src/ios/ios_deploy.dart

```diff
@@ -169,30 +211,193 @@ class IOSDeploy {
+
+        // (lldb)     run
+        // success
+        // 2020-09-15 13:42:25.185474-0700 Runner[477:181141] flutter: Observatory listening on http://127.0.0.1:57782/
+        if (line == '(lldb)     run') {
+          _logger.printTrace(line);
+          lldbRunCalled = true;
+          return;
+        }
+        // next line must be "success".
+        if (lldbRunCalled && line == 'success') {
+          _logger.printTrace(line);
+          _debuggerAttached = true;
+          debuggerCompleter.complete(true);
+          return;
+        }
+        lldbRunCalled = false;
+        if (line.contains('PROCESS_STOPPED') ||
+            line.contains('PROCESS_EXITED')) {
+          // The app exited or crashed, so stop echoing the output.
+          // Don't pass any further ios-deploy debugging messages to the log reader after it exits.
+          _debuggerAttached = false;
+          _logger.printTrace(line);
+          return;
+        }
+
+        if (line.contains('error: process launch failed')) {
+          // Timed out waiting for the app to launch.
+          _debuggerAttached = false;
+          _logger.printTrace(line);
+          debuggerCompleter.complete(false);
+          return;
+        }
+        if (!debuggerAttached) {
+          _logger.printTrace(line);
+          return;
+        }
+        if (lastLineFromDebugger != null && lastLineFromDebugger.isNotEmpty && line.isEmpty) {
+          // The lldb console stream from ios-deploy is separated lines by an extra \r\n.
+          // To avoid all lines being double spaced, if the last line from the
+          // debugger was not an empty line, skip this empty line.
+          // This will still cause "legit" logged newlines to be doubled...
+        } else {
+          _debuggerOutput.add(line);
+        }
+        lastLineFromDebugger = line;
+      });
+      final StreamSubscription<String> stderrSubscription = _iosDeployProcess.stderr
+          .transform<String>(utf8.decoder)
+          .transform<String>(const LineSplitter())
+          .listen((String line) {
+        _logger.printTrace(line);
+      });
+      unawaited(_iosDeployProcess.exitCode.then((int status) {
+        _logger.printTrace('ios-deploy exited with code $exitCode');
+        _debuggerAttached = false;
+        unawaited(stdoutSubscription.cancel());
+        unawaited(stderrSubscription.cancel());
+      }).whenComplete(() async {
+        if (_debuggerOutput.hasListener) {
+          // Tell listeners the process died.
+          await _debuggerOutput.close();
+        }
+        if (!debuggerCompleter.isCompleted) {
+          debuggerCompleter.complete(false);
+        }
+        _iosDeployProcess = null;
+      }));
+    } on ProcessException catch (exception, stackTrace) {
+      _logger.printTrace('ios-deploy failed: $exception');
+      _debuggerAttached = false;
+      _debuggerOutput.addError(exception, stackTrace);
+    } on ArgumentError catch (exception, stackTrace) {
```

Comment:
This will fail if ios-deploy is missing right? should we catch the error or let it crash here and solve the problem elsewhere? i.e. startup checks

Reply:
I think we already have those checks in place with the stamp/artifact download logic, since `ios-deploy` is an artifact we ship ourselves `idevicesyslog`.  Unlike the Xcode tooling where we do lots of install checks.